File: 300-DesksetBack/shared/log/log.go
package log

import (
	"fmt"
	"log"
	"os"
	"path/filepath"
	"time"

	"DesksetBack/shared/constant"
)

var (
	logFile *os.File
	logger  *log.Logger
)

type LogLevel int

const (
	INFO LogLevel = iota
	WARNING
	ERROR
	EXCEPTION
	PANIC
)

func openLogger() error {
	if err := os.MkdirAll(constant.LogDir, 0o755); err != nil {
		return fmt.Errorf("Failed to create log directory: %w", err)
	}

	logTime := time.Now().Format(constant.LogStemFormat)
	logPath := filepath.Join(constant.LogDir, constant.LogStemPrefix+logTime+"."+constant.LogExtn)

	var err error
	logFile, err = os.OpenFile(logPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
	if err != nil {
		return fmt.Errorf("Failed to create log file: %w", err)
	}

	logger = log.New(logFile, "", 0)
	return nil
}

func closeLogger() {
	if logFile != nil {
		logFile.Close()
	}
}

func SetupLogger() (func(), error) {
	if err := openLogger(); err != nil {
		return nil, err
	}
	return closeLogger, nil
}

func logMessage(level LogLevel, message string) {
	if logger == nil {
		if err := openLogger(); err != nil {
			log.Printf("Failed to open logger: %v", err)
			return
		}
	}

	timestamp := time.Now().Format("2006-01-02 15:04:05")
	levelStr := ""
	switch level {
	case INFO:
		levelStr = "INFO"
	case WARNING:
		levelStr = "WARNING"
	case ERROR:
		levelStr = "ERROR"
	case EXCEPTION:
		levelStr = "EXCEPTION"
	case PANIC:
		levelStr = "PANIC"
	default:
		levelStr = "UNKNOWN"
		return
	}

	logger.Printf("[%s] [%s]: %s", timestamp, levelStr, message)
}

func Info(message string) {
	logMessage(INFO, message)
}

func Warning(message string) {
	logMessage(WARNING, message)
}

func Error(message string) {
	logMessage(ERROR, message)
}

File: 350-DesksetBack.MCP/src/deskset/shared/log.py
from .constant import LOG_DIR, LOG_STEM_PREFIX, LOG_STEM_FORMAT, LOG_EXTN
import arrow

log_file = f'{LOG_STEM_PREFIX}{arrow.now().format(LOG_STEM_FORMAT)}.{LOG_EXTN}'


from pathlib import Path

Path(f'./{LOG_DIR}').mkdir(exist_ok=True)             # 创建日志目录，存在则跳过
Path(f'./{LOG_DIR}/{log_file}').touch(exist_ok=True)  # 创建日志文件，存在则跳过
# open(f'./{LOG_DIR}/{LOG_FILE}', 'w').close()          # 清空（上次）日志文件


import logging

logging.basicConfig(
    filename=f'./{LOG_DIR}/{log_file}',
    # filemode 注释
      # 注 1：目录不存在也会抛出 FileNotFoundError 异常
      # 注 2：用 a 模式写入，w 模式会意外覆盖运行时日志
    filemode='a',
    format='[%(asctime)s] [%(levelname)s]: %(message)s',
    datefmt='%Y-%m-%d %H:%M:%S',
    level=logging.INFO,
    encoding='utf-8'
)
